crates/ra_hir/src/code_model_impl/module.rs

```diff
@@ -126,15 +131,39 @@ impl Module {
         );
 
         let segments = &path.segments;
-        for name in segments.iter() {
+        for (idx, name) in segments.iter().enumerate() {
             let curr = if let Some(r) = curr_per_ns.as_ref().take_types() {
                 r
             } else {
                 return Ok(PerNs::none());
             };
             let module = match curr.resolve(db)? {
                 Def::Module(it) => it,
-                // TODO here would be the place to handle enum variants...
+                Def::Enum(e) => {
+                    if segments.len() == idx + 1 {
+                        // enum variant
+                        let matching_variant = e
+                            .variant_ids()
```

Comment:
there's a method returning the `EnumVariant`s directly, this will also be simpler :)